care/utils/notification_handler.py

```diff
@@ -144,23 +144,23 @@ def deserialize_extra_data(self, extra_data):
 
     def generate_extra_users(self):
         if isinstance(self.caused_object, PatientConsultation):
-            if self.caused_object.assigned_to:
-                self.extra_users.append(self.caused_object.assigned_to.id)
+            for clinician in self.caused_object.assigned_clinicians.all():
```

Comment:
```suggestion
            for clinician in self.caused_object.assigned_clinicians.all().only("id"):
```

we dont require all fields here so lets improve the query
Similar occurrences below V

Reply:
Interesting suggestion!

I was reading through https://docs.djangoproject.com/en/5.0/topics/db/optimization/#use-queryset-defer-and-only and was curious to profile the two queries, but while doing so, I am unable to use `.only` on `User` model.

(Trying these out via django shell)

```
PatientConsultation.objects.get(id=2268).assigned_clinicians.all().only("id")

FieldError: Field User.local_body cannot be both deferred and traversed using select_related at the same time.
Stacktrace: https://rentry.co/xourynxu/raw
```

The same error can also be replicated via `User.objects.all().only("id")`.

It appears that the `care.users.models.CustomUserManager` class is using local_body in `selected_related` 
```
class CustomUserManager(UserManager):
    def get_queryset(self):
        qs = super().get_queryset()
        return qs.filter(deleted=False, is_active=True).select_related(
            "local_body", "district", "state"
        )
```

Is there any way to work around this limitation of not being able to couple `.only` and `select_related`?

Reply:
Resolved after discussing in DMs.